xwiki-platform-core/xwiki-platform-security/xwiki-platform-security-requiredrights/xwiki-platform-security-requiredrights-default/src/main/java/org/xwiki/platform/security/requiredrights/internal/RequiredRightsEditConfirmationChecker.java

```diff
@@ -0,0 +1,97 @@
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.platform.security.requiredrights.internal;
+
+import java.util.Optional;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+
+import org.xwiki.component.annotation.Component;
+import org.xwiki.model.validation.edit.EditConfirmationChecker;
+import org.xwiki.model.validation.edit.EditConfirmationCheckerResult;
+import org.xwiki.platform.security.requiredrights.RequiredRightAnalyzer;
+import org.xwiki.platform.security.requiredrights.RequiredRightsException;
+import org.xwiki.script.ScriptContextManager;
+import org.xwiki.security.authorization.ContextualAuthorizationManager;
+import org.xwiki.security.authorization.Right;
+import org.xwiki.template.TemplateManager;
+
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.doc.XWikiDocument;
+
+import static javax.script.ScriptContext.GLOBAL_SCOPE;
+
+/**
+ * Check for the presence of required rights results for the current document before editing it.
+ *
+ * @version $Id$
+ * @since 15.9RC1
+ */
+@Component
+@Singleton
+@Named("requiredRights")
+public class RequiredRightsEditConfirmationChecker implements EditConfirmationChecker
+{
+    @Inject
+    @Named(XWikiDocumentRequiredRightAnalyzer.ID)
+    private RequiredRightAnalyzer<XWikiDocument> analyzer;
+
+    @Inject
+    private RequiredRightsChangedFilter requiredRightsChangedFilter;
+
+    @Inject
+    private Provider<XWikiContext> xcontextProvider;
+
+    @Inject
+    private TemplateManager templateManager;
+
+    @Inject
+    private ContextualAuthorizationManager authorization;
+
+    @Inject
+    private ScriptContextManager scriptContextManager;
+
+    @Override
+    public Optional<EditConfirmationCheckerResult> check()
+    {
+        XWikiContext context = this.xcontextProvider.get();
+        XWikiDocument tdoc = (XWikiDocument) context.get("tdoc");
+        if (!this.authorization.hasAccess(Right.EDIT, tdoc.getDocumentReferenceWithLocale())) {
+            return Optional.empty();
+        }
+        if (tdoc.isNew()) {
+            return Optional.empty();
+        }
+        try {
+            RequiredRightsChangedResult analysisResults =
+                this.requiredRightsChangedFilter.filter(tdoc.getAuthors(), this.analyzer.analyze(tdoc));
+            this.scriptContextManager.getCurrentScriptContext()
+                .setAttribute("analysisResults", analysisResults, GLOBAL_SCOPE);
+            return Optional.of(new EditConfirmationCheckerResult(
+                this.templateManager.executeNoException("security/requiredrights/requiredRightsConfirmationChecker.vm",
+                    false), false));
+        } catch (RequiredRightsException e) {
+            throw new RuntimeException(e);
```

Comment:
I think I would have handled it with an actual EditConfirmationCheckerResult returning an error with the root cause of the exception. Basically saying that it could be dangerous to edit since the analysis stopped.

Reply:
indeed, good idea, thanks

Reply:
Check also the discussion on https://github.com/xwiki/xwiki-platform/pull/2470#discussion_r1363447951